test_pull_requests

```diff
@@ -1719,22 +1722,24 @@ popd
           pretest_settings = Properties['settings'][settings['pretest_settings_key']]
           repo_to_comment_fields = {}
           can_skip_merge_tests = false
-          if pretest_settings && repo_to_pull_request.length > 0
+          if trigger_trusted && pretest_settings && repo_to_pull_request.length > 0
             pretest_prefix = pretest_settings['test_prefix']
-            can_skip_merge_tests = repo_to_pull_request.all? do |repo, pull_request|
-              pretest_comment = get_comment_with_prefix(pull_request['number'], repo, pretest_prefix, (repo == base_repo ? comments : nil))
+            can_skip_merge_tests = true
+            repo_to_pull_request.each do |repo, sub_pull_request|
+              pretest_comment = get_comment_with_prefix(sub_pull_request['number'], repo, pretest_prefix, (repo == base_repo ? comments : nil))
+              # Don't skip merge tests if the user isn't trusted to merge this coreq PR
               if pretest_comment.nil?
-                false
+                can_skip_merge_tests = false
               else
                 pretest_fields = extract_bot_comment_fields(pretest_comment['body'], pretest_settings)
-                pretest_pr_base_commit = base_repo_commit_for_pull_req(pull_request)
+                pretest_pr_base_commit = base_repo_commit_for_pull_req(sub_pull_request)
                 can_skip_one = (pretest_fields[:state] == :success &&
                                 pretest_fields[:base_commit] == pretest_pr_base_commit['sha'])
                 if can_skip_one
                   # Store pre-merge comment fields to carry the build url and base commit id through to SUCCESS
                   repo_to_comment_fields[repo] = {:state => :success, :build_url => pretest_fields[:build_url], :base_commit => pretest_fields[:base_commit]}
                 end
-                can_skip_one
+                can_skip_merge_tests &= can_skip_one
```

Comment:
Is this all to figure out if you can skip?  If so, should there be a break and none of this &= logic?